addons/survey/tests/test_survey_security.py

```diff
@@ -140,25 +140,34 @@ def test_access_survey_survey_manager(self):
     @mute_logger('odoo.addons.base.models.ir_model')
     @users('survey_user')
     def test_access_survey_survey_user(self):
-        # Create: own only
-        survey = self.env['survey.survey'].create({'title': 'Test Survey 2'})
-        self.env['survey.question'].create({'title': 'My Page', 'sequence': 0, 'is_page': True, 'question_type': False, 'survey_id': survey.id})
-        self.env['survey.question'].create({'title': 'My Question', 'sequence': 1, 'survey_id': survey.id})
-
-        # Read: all
+        # Restrict common survey to survey_manager
+        restricted_to_other_survey = self.survey
+        restricted_to_other_survey.write({'restrict_user_ids': [[4, restricted_to_other_survey.user_id.id]]})
+
+        # Create: restricted to self or no one
+        unrestricted_survey = self.env['survey.survey'].create({'title': 'Test Survey Unrestricted'})
+        self.env['survey.question'].create({'title': 'My Page', 'sequence': 0, 'is_page': True, 'question_type': False, 'survey_id': unrestricted_survey.id})
+        self.env['survey.question'].create({'title': 'My Question', 'sequence': 1, 'survey_id': unrestricted_survey.id})
+        restricted_to_self_survey = self.env['survey.survey'].create({'title': 'Test Survey Restricted to Self', 'restrict_user_ids': [[4, self.env.user.id]]})
+        with self.assertRaises(ValidationError):
+            self.env['survey.survey'].with_user(self.env.user).create({
+                'title': 'Test Survey Restricted to Other', 'restrict_user_ids': [[4, restricted_to_other_survey.user_id.id]]})
+
+        # Read: restricted to self or no one
         surveys = self.env['survey.survey'].search([('title', 'ilike', 'Test')])
-        self.assertEqual(surveys, self.survey | survey)
+        self.assertEqual(surveys, restricted_to_self_survey | unrestricted_survey)
         surveys.read(['title'])
 
-        # Write: own only
-        survey.write({'title': 'New Title'})
+        # Write: restricted to self or no one
+        unrestricted_survey.write({'title': 'New Title'})
+        restricted_to_self_survey.write({'title': 'New Title'})
```

Comment:
Could group operation
```suggestion
        (unrestricted_survey + restricted_to_self_survey).write({'title': 'New Title'})
```